pkg/agent/controller/egress/egress_controller.go

```diff
@@ -1118,39 +1118,42 @@ func (c *EgressController) syncEgress(egressName string) error {
 	}()
 
 	egressIP := net.ParseIP(eState.egressIP)
-	// Install SNAT flows for desired Pods.
-	for pod := range pods {
-		eState.pods.Insert(pod)
-		stalePods.Delete(pod)
+	egress, _ = c.egressLister.Get(egressName)
```

Comment:
can't we use the current `egress` object here, why do we need to get it from the cache again?